pkg/controller/compliancescan/compliancescan_controller_test.go

```diff
@@ -78,6 +80,52 @@ var _ = Describe("Testing compliancescan controller phases", func() {
 			Expect(compliancescaninstance.Status.Phase).To(Equal(compv1alpha1.PhaseLaunching))
 			Expect(compliancescaninstance.Status.Result).To(Equal(compv1alpha1.ResultNotAvailable))
 		})
+
+		Context("With correct custom RawResultStorageSize", func() {
+			It("should update the compliancescan instance to phase LAUNCHING", func() {
+				compliancescaninstance.Spec.RawResultStorageSize = "2Gi"
+				result, err := reconciler.phasePendingHandler(compliancescaninstance, logger)
+				Expect(result).NotTo(BeNil())
+				Expect(err).To(BeNil())
+				Expect(compliancescaninstance.Status.Phase).To(Equal(compv1alpha1.PhaseLaunching))
+				Expect(compliancescaninstance.Status.Result).To(Equal(compv1alpha1.ResultNotAvailable))
+			})
+		})
+
+		Context("With missing RawResultStorageSize", func() {
+			It("should update the compliancescan instance with the default size", func() {
+				compliancescaninstance.Spec.RawResultStorageSize = ""
+				result, err := reconciler.phasePendingHandler(compliancescaninstance, logger)
+				Expect(result).NotTo(BeNil())
+				Expect(err).To(BeNil())
+
+				scan := &compv1alpha1.ComplianceScan{}
+				key := types.NamespacedName{
+					Name:      compliancescaninstance.Name,
+					Namespace: compliancescaninstance.Namespace,
+				}
+				reconciler.client.Get(context.TODO(), key, scan)
+				Expect(scan.Spec.RawResultStorageSize).To(Equal(compv1alpha1.DefaultRawStorageSize))
+			})
+		})
+
+		Context("With invalid RawResultStorageSize", func() {
+			It("report an error and move to phase DONE", func() {
+				compliancescaninstance.Spec.RawResultStorageSize = "invalid"
```

Comment:
Would it be worth testing a scenario where you gave a valid size but one that the cluster can't accommodate due to other limits? i.e., 1000000Gi

Reply:
That's a good point! Let me test it out locally and add a test case for that.

Reply:
Should probably output an error too.

Reply:
Done. Added an e2e test for that.